src/piadm.sh

```diff
@@ -836,12 +834,12 @@ update_CN() {
 		err "The update command may only be used on a Triton Compute Node"
 	fi
 
-	piname_present_get_bootfs "ipxe" $1
-	cd /${bootfs}
+	piname_present_get_bootfs "ipxe" "$1"
+	cd "/${bootfs}" || fatal "Could not chdir to /$bootfs"
 
 	# First check if the backup PI is in need of update.
 	# The standard iPXE/CN deployment has exactly one platform-STAMP.
-	mapfile -t pdirs < <(ls | grep -v ipxe | grep platform-)
+	mapfile -t pdirs < <(ls -d platform-[0-9]*Z)
```

Comment:
I don't think this is going to match, because the pattern is `20[0-9]{6}T[0-9]{6}Z`. I.e., you're going to miss the `T`

Reply:
Shell globs aren't proper regexes.
```
[root@larry (kebecloud) ~]# pargs $$
128610:	-bash
argv[0]: -bash
[root@larry (kebecloud) ~]# ls -d /zones/boot/platform-[0-9]*Z
/zones/boot/platform-20201002T173911Z
[root@larry (kebecloud) ~]# 
```
it'll match.

Reply:
Fixed with platform-[0-9]*T*Z